contracts/mocks/TokenProxy.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+// Original contract: https://github.com/convex-eth/platform/blob/main/contracts/contracts/wrappers/ConvexStakingWrapper.sol
+pragma solidity 0.8.6;
+import "@yield-protocol/utils-v2/contracts/token/ERC20.sol";
+
+contract TokenProxy {
+    ERC20 immutable _curve;
+
+    constructor(ERC20 curve_) {
+        _curve = curve_;
+    }
+
+    /// @param _owner The address from which the balance will be retrieved
+    /// @return balance the balance
+    function balanceOf(address _owner) external view returns (uint256 balance) {
+        return _curve.balanceOf(_owner);
+    }
+
+    /// @notice send `_value` token to `_to` from `msg.sender`
+    /// @param _to The address of the recipient
+    /// @param _value The amount of token to be transferred
+    /// @return success Whether the transfer was successful or not
+    function transfer(address _to, uint256 _value) external returns (bool success) {
+        return _curve.transfer(_to, _value);
+    }
+
+    /// @notice send `_value` token to `_to` from `_from` on the condition it is approved by `_from`
+    /// @param _from The address of the sender
+    /// @param _to The address of the recipient
+    /// @param _value The amount of token to be transferred
+    /// @return success Whether the transfer was successful or not
+    function transferFrom(
+        address _from,
+        address _to,
+        uint256 _value
+    ) external returns (bool success) {
+        return _curve.transferFrom(_from, _to, _value);
+    }
+
+    /// @notice `msg.sender` approves `_addr` to spend `_value` tokens
+    /// @param _spender The address of the account able to transfer the tokens
+    /// @param _value The amount of wei to be approved for transfer
+    /// @return success Whether the approval was successful or not
+    function approve(address _spender, uint256 _value) external returns (bool success) {
+        return _curve.approve(_spender, _value);
```

Comment:
Same as above, `msg.sender` is different. Here I don't think that there is a workaround.

Reply:
As above